### PR TITLE
Add documentation for Scaffold-ETH

### PR DESCRIPTION
**Pull request by hyd628**
### Description

Adds a new page on how to set up Scaffold-ETH to use with Moonbeam networks. 

### Checklist

- [x] If this requires translations for the `moonbeam-docs-cn` repo, I have created a ticket for the translations in Jira
- [x] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects
- [x] If images have been added, I have run the `compress-images.py` script to compress the images.
- [x] If images have been moved, I have created an additional PR in `moonbeam-docs-cn` to update images to their new paths
- [x] If variables (in variables.yml) need to be updated (such as a name change), I have updated the `moonbeam-docs-cn` repo to use the new variables

### Corresponding PRs

Please link to any corresponding PRs here.

### After Translation Requirements

- [ ] Will need to create PR in `moonbeam-docs` repo to remove images
- [ ] Will need to create PR in `moonbeam-docs` repo to remove variables
- [ ] Will need to create PR in `moonbeam-mkdocs` repo to add redirects for Chinese site
- [ ] No additional PRs are required after the translations are done

#### Items to be Updated

Please list any of the items that will need to be added or deleted after the translations are done here.

### CODE REVIEW

File: builders/interact/scaffold-eth.md [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+
+## Modify Configurations {: #modify-configurations }
+
+We need to make modifications to the configurations of the three major components that make up Scaffold-ETH.
+
+### Hardhat Component {: #hardhat-component }
+
+Let's begin with making some modifications to the Hardhat component under the `/packages/hardhat` folder.
+
+1. The file we need to modify is `scaffold-eth/packages/hardhat/hardhat.config.js`. Under the `module.exports/networks` section, we want to add the network definitions for the Moonbeam network(s) that we would like to use, including the RPC endpoint, the chain ID, and the contract deployment account's private key.
```
Comment (eshaben) [inline builders/interact/scaffold-eth.md:54]: For all of the numbered list items throughout, no period is required at the end of the sentences

File: builders/interact/scaffold-eth.md [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+            accounts: ["Deployment Account Private Key"]
+        }
+        ```
+
+    === "Moonbeam Dev Node"
+        ```js
+        moonbeamDevNode: {
+            url: 'http://127.0.0.1:9933',
+            chainId: 1281,  // 0x501 in hex,
+            accounts: ["Deployment Account Private Key"]
```
Comment (eshaben) [inline builders/interact/scaffold-eth.md:88]: can you please use the variables for the url and chainIds here and throughout the guide?

**Review by eshaben — CHANGES_REQUESTED** [review]
I really love the layout of this guide! Nice job!! Just some minor things 🙂

**Comment by hyd628** [comment]
Thanks @eshaben . I have incorporated all the feedback.

**Review by eshaben — APPROVED** [review]
LGTM 🚀

**Pull request merged**